Modules/cjkcodecs/multibytecodec.c

```diff
@@ -1670,6 +1670,9 @@ _multibytecodec_MultibyteStreamWriter_writelines(MultibyteStreamWriterObject *se
         if (r == -1)
             return NULL;
     }
+    /* PySequence_Length() can fail */
+    if (PyErr_Occurred())
```

Comment:
Maybe cache `PySequence_Length`. It's not free. And even `lines` is changed  the behaviour is not bad.